src/test/com/fulcrologic/rad/database_adapters/sql_test.clj

```diff
@@ -0,0 +1,70 @@
+(ns com.fulcrologic.rad.database-adapters.sql-test
+  (:require
+    [com.fulcrologic.rad.attributes            :as rad.attr :refer [defattr]]
+    [com.fulcrologic.rad.database-adapters.sql :as rad.sql]
+    [clojure.test :refer [deftest testing is]]))
+
+(defattr account-id :account/id :uuid
+  {::rad.attr/identity? true
+   ::rad.sql/schema :production
+   ::rad.sql/tables #{"accounts"}})
+
+(defattr account-name :account/name :string
+  {::rad.sql/schema :production
+   ::rad.sql/tables #{"accounts"}})
+
+;; Derived data
+(defattr account-locked? :account/locked? :boolean
+  {})
+
+(defattr user-id :user/id :uuid
+  {::rad.sql/schema :production
+   ::rad.sql/tables #{"users"}})
+
+(defattr user-name :user/name :string
+  {::rad.sql/schema :production
+   ::rad.sql/tables #{"users"}})
+
+(deftest delta->txs-test
+  (with-redefs [rad.attr/key->attribute {:account/id      account-id
+                                         :account/name    account-name
+                                         :account/locked? account-locked?
+                                         :user/id         user-id
+                                         :user/name       user-name}]
+
+    (let [account-id #uuid "ffffffff-ffff-ffff-ffff-000000000001"
```

Comment:
This is kind of a "dumping ground" style of testing (too many things going on in a single test). I generally discourage this because:

1. It tends to lose specificity. "It returns a list of updates", for example, is actually covering checks for all kinds of things: the table is in the update, its type is an update, the way things are encoded, etc.  A failure requires a lot of understanding of the underlying API and isn't immediately obvious.
2. It is harder to read. The follow-on developer has to read a lot of stuff to tease apart things, which creates more drag during later development.
3. It complects a lot of behaviors. The code is doing a number of different things, and the assertions are lumping them together. 

Consider that what your logic needs to do is figure out the correct SQL instruction for various cases:

1. before is empty, after has a value
2. opposite of 1
3. both have values that don't match
4. both have values that match
5. The attribute itself isn't "interesting" (i.e. is for a different store)

You've encoded most of these interesting behaviors into the maps instead of into the assertions.

Reply:
I didn't get the entire point, maybe you could explain some things in our next call. I've added these tests for myself to be able to make changes quicker, and just may remove them. But I'm happy to learn some of these ideas.

Reply:
I was just enumerating individual behaviors, as I see them...I.e. the delta has no `:before` but it does have an `:after`, etc.

Reply:
Ah I see, makes sense.

Reply:
those look better :)